nav2_map_server/src/vo_server/vector_object_server.cpp

```diff
@@ -0,0 +1,543 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_map_server/vector_object_server.hpp"
+
+#include <chrono>
+#include <exception>
+#include <functional>
+#include <limits>
+#include <stdexcept>
+#include <utility>
+
+#include "rclcpp/create_timer.hpp"
+
+#include "tf2_ros/create_timer_ros.hpp"
+
+#include "nav2_util/occ_grid_utils.hpp"
+#include "nav2_util/occ_grid_values.hpp"
+
+using namespace std::placeholders;
+
+namespace nav2_map_server
+{
+
+VectorObjectServer::VectorObjectServer(const rclcpp::NodeOptions & options)
+: nav2::LifecycleNode("vector_object_server", "", options), process_map_(false)
+{}
+
+nav2::CallbackReturn
+VectorObjectServer::on_configure(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(get_logger(), "Configuring");
+
+  // Transform buffer and listener initialization
+  tf_buffer_ = std::make_shared<tf2_ros::Buffer>(this->get_clock());
+  auto timer_interface = std::make_shared<tf2_ros::CreateTimerROS>(
+    this->get_node_base_interface(),
+    this->get_node_timers_interface());
+  tf_buffer_->setCreateTimerInterface(timer_interface);
+  tf_listener_ = std::make_shared<tf2_ros::TransformListener>(*tf_buffer_);
+
+  // Obtaining ROS parameters
+  if (!obtainParams()) {
+    return nav2::CallbackReturn::FAILURE;
+  }
+
+  map_pub_ = create_publisher<nav_msgs::msg::OccupancyGrid>(
+    map_topic_,
+    nav2::qos::LatchedPublisherQoS());
+
+  add_shapes_service_ = create_service<nav2_msgs::srv::AddShapes>(
+    "~/add_shapes",
+    std::bind(&VectorObjectServer::addShapesCallback, this, _1, _2, _3));
+
+  get_shapes_service_ = create_service<nav2_msgs::srv::GetShapes>(
+    "~/get_shapes",
+    std::bind(&VectorObjectServer::getShapesCallback, this, _1, _2, _3));
+
+  remove_shapes_service_ = create_service<nav2_msgs::srv::RemoveShapes>(
+    "~/remove_shapes",
+    std::bind(&VectorObjectServer::removeShapesCallback, this, _1, _2, _3));
+
+  return nav2::CallbackReturn::SUCCESS;
+}
+
+nav2::CallbackReturn
+VectorObjectServer::on_activate(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(get_logger(), "Activating");
+
+  map_pub_->on_activate();
+
+  // Trigger map to be published
+  process_map_ = true;
+  switchMapUpdate();
+
+  // Creating bond connection
+  createBond();
+
+  return nav2::CallbackReturn::SUCCESS;
+}
+
+nav2::CallbackReturn
+VectorObjectServer::on_deactivate(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(get_logger(), "Deactivating");
+
+  // Destroying bond connection
+  destroyBond();
```

Comment:
This should be the last thing in the deactivate function, move to right before the return statement